wrangler-core/src/main/java/io/cdap/wrangler/utils/StructuredToRowTransformer.java

```diff
@@ -77,14 +78,36 @@ public static Object getValue(StructuredRecord input, String fieldName) {
       }
     }
 
-    // if the type is bytes, need to make sure the value is byte array since byte buffer is not serializable
-    if (fieldSchema.getType().equals(Schema.Type.BYTES)) {
-      Object val = input.get(fieldName);
-      return val instanceof ByteBuffer ? Bytes.toBytes((ByteBuffer) val) : val;
-    }
+    Object val = input.get(fieldName);
+    return processValue(val, fieldSchema);
+  }
 
-    // If the logical type is present in complex types, it will be retrieved as corresponding
-    // simple type (int/long).
-    return input.get(fieldName);
+  private static Object processValue(Object val, Schema schema) {
+    switch(schema.getType()) {
+      // if the type is bytes, need to make sure the value is byte array since byte buffer is not serializable
+      case BYTES:
+        return val instanceof ByteBuffer ? Bytes.toBytes((ByteBuffer) val) : val;
+      // Recursively process structured records.
+      case RECORD:
+        return val instanceof StructuredRecord ? transform((StructuredRecord) val) : val;
```

Comment:
what other type can `val` have here? Should we throw exception if it is not a StructuredRecord?